frappe/utils/response.py

```diff
@@ -112,22 +112,15 @@ def as_json():
 def as_pdf():
 	response = Response()
 	response.mimetype = "application/pdf"
-	encoded_filename = quote(frappe.response["filename"].replace(" ", "_"))
-	response.headers["Content-Disposition"] = (
-		'filename="%s"' % frappe.response["filename"].replace(" ", "_")
-		+ ";filename*=utf-8''%s" % encoded_filename
-	).encode("utf-8")
+	response.headers.add("Content-Disposition", filename=frappe.response["filename"])
 	response.data = frappe.response["filecontent"]
 	return response
 
 
 def as_binary():
 	response = Response()
 	response.mimetype = "application/octet-stream"
-	filename = "_".join(frappe.response["filename"].split())
-	response.headers["Content-Disposition"] = (
-		'filename="%s"' % filename
-	).encode("utf-8")
+	response.headers.add("Content-Disposition", filename=frappe.response["filename"])
 	response.data = frappe.response["filecontent"]
 	return response
```

Comment:
`as_pdf` and `as_binary` seem to be broken.. `headers.add` requires a "value". perhaps you should specify the default value "inline".